mirrord/operator/src/client.rs

```diff
@@ -628,39 +548,39 @@ impl OperatorApi<PreparedClientCert> {
     {
         self.check_feature_support(config)?;
 
-        // Check if the deployment is empty if so then
-        let is_empty_deployment = match &config.target.path {
-            Some(Target::Deployment(target)) => {
-                let deployment_api = get_k8s_resource_api::<Deployment>(
-                    &self.client,
-                    config.target.namespace.as_deref(),
-                );
-
-                let deployment = deployment_api
-                    .get(&target.deployment)
-                    .await
-                    .map_err(|error| OperatorApiError::KubeError {
-                        error,
-                        operation: OperatorOperation::FindingTarget,
-                    })?;
-
-                !deployment
-                    .status
-                    .map(|status| status.available_replicas > Some(0))
-                    .unwrap_or_default()
-            }
-            _ => false,
-        };
+        if target.containers_count() > 1
+            && config
+                .target
+                .path
+                .as_ref()
+                .is_some_and(|target| target.container().is_none())
+        {
```

Comment:
Warning logic should match the logic when running without an operator - don't warn if there is only 1 non-skipped container (skipped containers are e.g. containers inserted by service meshes etc.).
If those two cases could share code for the common logic that could be nice.

Reply:
Got it.

How far should I go to merge these apis? I spent some time playing around with it, and I think it would be nice if we leaned towards using `ResolvedTarget` more. What I mean is, create a type for each variant, then implemented the `RuntimeData*` stuff that we have for `config::Target` for the variants.

The quickest approach would be to just pass the `config::Target` to the `connect_in_new_session`, and call `target.runtime_data(operator_client, ...)`.

The end result will be pretty much the same, but if we implement this in `ResolvedTarget`, we kinda decouple our code more from config stuff, and I think it would look a bit less confusing.

:thinking: :thought_balloon:  Thoughts?

Reply:
Didn't completely understand, but IMO the best thing for this PR would be to stay with the current behaviour and logic, especially if it does not mean writing a lot of new code. I think it's best to separate refactors from actual changes when possible, and only introduce new stuff together with refactors if keeping the old logic would mean writing new code that would immediately be replaced on the change to the new logic.
If this is the case here (keeping old logic would require a significant amount of new code), let's do call and discuss those options while looking at the code together.

Reply:
Alright, now I think we're using the right stuff to check if we're in multi-container.